### PR TITLE
feat: mechanism for registering pipelines that can be used by cli

### PR DESCRIPTION
**Pull request by elonp**

### CODE REVIEW

**Review by jzazo — APPROVED** [review]

File: rats-processors/test/python/rats_test/pipelines_app/example/_pipeline2.py [hunk 1]
```diff
@@ -60,3 +60,11 @@ def p2(self) -> ux.Pipeline[P2Inputs, P2Outputs]:
         p2 = self.get(apps.method_service_id(self.train_model))
         p = self.combine([p1, p2], dependencies=[p1.outputs.data >> p2.inputs.data])
         return cast(ux.Pipeline[P2Inputs, P2Outputs], p)
+
+    @apps.group(rpa.PipelineRegistryGroups.EXECUTABLE_PIPELINES)
```
Comment (jzazo) [inline rats-processors/test/python/rats_test/pipelines_app/example/_pipeline2.py:64]: How about we have a decorator `executable_pipeline` that takes a name and doc and is used on top of the pipeline provider method?

File: rats-processors/test/python/rats_test/pipelines_app/example/_pipeline2.py [hunk 2]
```diff
@@ -60,3 +60,11 @@ def p2(self) -> ux.Pipeline[P2Inputs, P2Outputs]:
...
+    def executable_pipeline(self) -> Any:
+        yield dict(
+            name="p2",
+            doc="Example pipeline 2",
+            service_id=apps.method_service_id(self.p2),
```
[the remark this answers is not in the document]
Comment (elonp) [reply rats-processors/test/python/rats_test/pipelines_app/example/_pipeline2.py:69]: @ms-lolo Is it OK for this method to yield?  I defined rpa.PipelineRegistryGroups.EXECUTABLE_PIPELINES as a ServiceId[Iterable[..]].  I am not sure yield and Iterable play nicely with caching.

File: rats-processors/src/python/rats/pipelines_app/_pipeline_registry.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+from rats.processors import ux
+
+
+class PipelineRegistryEntry(TypedDict):
+    name: str
+    doc: str
+    service_id: apps.ServiceId[ux.UPipeline]
+
+
+class PipelineRegistryGroups:
```
[the remark this answers is not in the document]
Comment (elonp) [reply rats-processors/src/python/rats/pipelines_app/_pipeline_registry.py:14]: @ms-lolo 
I think we need a standard way to name these service id "enum" classes, and it should apply to services, groups, configs, etc.

So do we go for:
* PipelineRegistryGroup / PipelineRegistryService
* PipelineRegistryGroups / PipelineRegistryServices
* PipelineRegistryGroupId / PipelineRegistryServiceId
* PipelineRegistryGroupId / PipelineRegistryId
* PipelineRegistryGroupIds / PipelineRegistryServiceIds
* PipelineRegistryGroupIds / PipelineRegistryIds

Let's vote on:
* singular / plural
* with / without Id suffix
* Use Service suffix for services or only for Groups/Configs

[the remark this answers is not in the document]
Comment (elonp) [reply rats-processors/src/python/rats/pipelines_app/_pipeline_registry.py:14]: I vote:
* plural
* no Id suffix
* with Service

**Pull request merged**